WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+			$sniff_helper['post_pagination'] = true;
+		}
+	}
+
+	/**
+	 * Verify that get_post_format()or has_format() are found, at least once
+	 * if the theme has a add_theme_support( 'post-format' ) call. This should
+	 * become an error if the theme is tagged with post-formats.
+	 * If one is present set flag to true for use later.
+	 *
+	 *  @param string $file_content contains content of file in a single string.
+	 */
+	public function get_post_format_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'get_post_format' ) ) {
+			$sniff_helper['post_format_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'has_post_format' ) ) {
+			$sniff_helper['post_format_support'] = true;
+		}
+	}
+
+	/**
+	 * Verify that the_post_thumbnail(), get_the_post_thumbnail(), or
+	 * get_post_thumbnail_id() are found at least once if the theme has a
+	 * add_theme_support( 'post-thumbnails' ) call. This should become an error
+	 * if the theme is tagged with featured-image.
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_post_thumbnail_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'the_post_thumbnail' ) ) {
+			$sniff_helper['post_thumbnail_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_the_post_thumbnail' ) ) {
+			$sniff_helper['post_thumbnail_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_post_thumbnail_id' ) ) {
+			$sniff_helper['post_thumbnail_support'] = true;
+		}
+	}
+
+	/**
+	 * Check that post tags are supported in the theme. At least one of the
+	 * following functions would need to be found in at least one of the template
+	 * files, fail if none are found at all; the_tags(), get_the_tag_list(),
+	 * get_the_term_list()
+	 * If one is present set flag to true for use later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_post_tags_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== strpos( $file_content, 'the_tags' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_the_tag_list' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'get_the_term_list' ) ) {
+			$sniff_helper['post_tags_support'] = true;
+		}
+	}
+
+	/**
+	 * Check that add_theme_support( 'title-tag' ) and wp_title() are being used.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_title_tag_check( $file_content ) {
+		global $sniff_helper;
+		if ( ! preg_match( '#add_theme_support\s?\(\s?[\'|"]title-tag#', $file_content ) ) {
+			$sniff_helper['title_tag']['theme_support'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_title' ) ) {
```

Comment:
We need to be careful here as this check will trip if someone is using the filter `wp_title` which is fine. Using the function `wp_title()` is not.